x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -422,6 +410,108 @@ public void testIsNullPushdownFilter() {
+                "!=" + testCase.tooLow,
+                "!= 0.0/0.0"
+            );
+            List<String> alwaysFalsePredicates = List.of(
+                "<" + testCase.tooLow,
+                "<=" + testCase.tooLow,
+                ">" + testCase.tooHigh,
+                ">=" + testCase.tooHigh,
+                "==" + testCase.tooHigh,
+                "==" + testCase.tooLow,
+                "< 0.0/0.0",
+                "<= 0.0/0.0",
+                "> 0.0/0.0",
+                ">= 0.0/0.0",
+                "== 0.0/0.0"
+            );
+
+            for (String truePredicate : trueForSingleValuesPredicates) {
+                String comparison = testCase.fieldName + truePredicate;
+                var query = "from test | where " + comparison;
+                Source expectedSource = new Source(1, 18, comparison);
+
+                EsQueryExec actualQueryExec = doTestOutOfRangeFilterPushdown(query);
+
+                assertThat(actualQueryExec.query(), is(instanceOf(SingleValueQuery.Builder.class)));
+                var actualLuceneQuery = (SingleValueQuery.Builder) actualQueryExec.query();
+                assertThat(actualLuceneQuery.field(), equalTo(testCase.fieldName));
+                assertThat(actualLuceneQuery.source(), equalTo(expectedSource));
+
+                assertThat(actualLuceneQuery.next(), equalTo(QueryBuilders.matchAllQuery()));
+            }
+
+            for (String falsePredicate : alwaysFalsePredicates) {
+                String comparison = testCase.fieldName + falsePredicate;
+                var query = "from test | where " + comparison;
+                Source expectedSource = new Source(1, 18, comparison);
+
+                EsQueryExec actualQueryExec = doTestOutOfRangeFilterPushdown(query);
+
+                assertThat(actualQueryExec.query(), is(instanceOf(SingleValueQuery.Builder.class)));
+                var actualLuceneQuery = (SingleValueQuery.Builder) actualQueryExec.query();
+                assertThat(actualLuceneQuery.field(), equalTo(testCase.fieldName));
+                assertThat(actualLuceneQuery.source(), equalTo(expectedSource));
+
+                var expectedInnerQuery = QueryBuilders.boolQuery().mustNot(QueryBuilders.matchAllQuery());
+                assertThat(actualLuceneQuery.next(), equalTo(expectedInnerQuery));
+            }
+        }
+    }
+
+    /**
+     * Expects e.g.
+     * LimitExec[500[INTEGER]]
+     * \_ExchangeExec[[],false]
+     *   \_ProjectExec[[!alias_integer, boolean{f}#190, byte{f}#191, constant_keyword-foo{f}#192, date{f}#193, double{f}#194, ...]]
+     *     \_FieldExtractExec[!alias_integer, boolean{f}#190, byte{f}#191, consta..][]
+     *       \_EsQueryExec[test], query[{"esql_single_value":{"field":"byte","next":{"match_all":{"boost":1.0}},...}}]
+     */
+    private EsQueryExec doTestOutOfRangeFilterPushdown(String query) {
+        var analyzer = makeAnalyzer("mapping-all-types.json", new EnrichResolution());
```

Comment:
Here you are creating a new Analyzer for every method call and this is actually not needed.